sound/soc/sof/ipc4-loader.c

```diff
@@ -342,9 +342,38 @@ static int sof_ipc4_load_library(struct snd_sof_dev *sdev, struct snd_sof_widget
 	return ret;
 }
 
+static int sof_ipc4_reload_libraries(struct snd_sof_dev *sdev)
+{
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
+	int i, ret;
+
+	/* Index 0 is reserved for base firmware */
+	for (i = 1; i < ipc4_data->max_fw_libs; i++) {
+		if (!ipc4_data->module_lib_info[i].id)
+			break;
+
+		if (sof_ops(sdev)->load_library) {
+			ret = sof_ops(sdev)->load_library(sdev, &ipc4_data->module_lib_info[i]);
+			if (ret < 0) {
+				dev_err(sdev->dev, "Failed loading FW library for module: %s\n",
+					ipc4_data->module_lib_info[i].name);
+				return ret;
+			}
+
+			dev_dbg(sdev->dev, "Loaded FW library for module: %s\n",
+				ipc4_data->module_lib_info[i].name);
+		} else {
+			return -EINVAL;
```

Comment:
the test on the load_library can be made outside of the loop.

But the question is how many times do we need to test for the existing of load_library? Some of these tests are probably overkill or always-true.